engine/server.go

```diff
@@ -0,0 +1,97 @@
+package engine
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"time"
+)
+
+type LogLevel int
+
+const (
+	LogDebug LogLevel = iota
+	LogInfo
+	LogWarn
+	LogError
+)
+
+type Logger interface {
+	Errorf(fmt string, v ...interface{})
+	Warnf(fmt string, v ...interface{})
+	Infof(fmt string, v ...interface{})
+	Debugf(fmt string, v ...interface{})
+}
+
+type Options func(*Server)
+
+func OptionPingInterval(time.Duration) Options                      { return nil }
+func OptionPingTimeout(time.Duration) Options                       { return nil }
+func OptionMaxBufferSize(int) Options                               { return nil }
+func OptionLogLevel(level LogLevel) Options                         { return nil }
+func OptionLogger(logger Logger) Options                            { return nil }
+func OptionTransports(initial string, upgradings ...string) Options { return nil }
+func OptionJSONP(padding int) Options                               { return nil }
```

Comment:
i think `padding` must be types value

Reply:
`padding` is an integer, see `j` in https://github.com/socketio/engine.io-protocol#jsonp.

Reply:
aaaa. ok, seem good. sorry